### DISCUSSION TITLE
Lesson 12: Deployment artifacts

### DISCUSSION THREAD

**Discussion by NathanPtrs** [smartcontractkit/full-blockchain-solidity-course-py#847] [Q&A] [answered]
Running "01_deploy_box.py" works fine. But follow up with "02_upgrade_box.py" generates the following error:

 File "C:\Users\Computer\AppData\Roaming\Python\Python310\site-packages\brownie\_cli\run.py", line 49, in main
    return_value, frame = run(
  File "C:\Users\Computer\AppData\Roaming\Python\Python310\site-packages\brownie\project\scripts.py", line 103, in run
    return_value = f_locals[method_name](*args, **kwargs)
  File ".\scripts\02_upgrade_box.py", line 20, in main
    proxy = TransparentUpgradeableProxy[-1]
  File "C:\Users\Computer\AppData\Roaming\Python\Python310\site-packages\brownie\network\contract.py", line 160, in __getitem__
    return self._contracts[i]
IndexError: list index out of range

TransparentUpgradeableProxy[-1] is failing because it lost that data between runs. I've narrowed it down to having something to do with deployment artifacts. Tried adding "dev_deployment_artifacts: True" in brownie-config.yaml and running Ganache Gui before the scripts so that both scripts access the same blockchain data but no luck. The only way I can make it work is by running the deploy script with "-I" and then, in the resulting console prompt:   run("scripts/02_upgrade_box.py")

It works without a problem on Rinkeby, so this is only happening in local development environments.

Might be more of a Brownie question but it does happen during the course...and I have a feeling I'll need an answer to this situation sooner or later.

**Comment by cromewar** [comment, marked as answer]
Hello @NathanPtrs, yes you are right this is more a problem with brownie/python rather the contract themselfs, using `-1` tries to locate the latest compilation on the build folder, however this does not work sometimes, I strongly suggest to delete the build folder and try again, that might help.

**Reply by NathanPtrs** [reply]
I tried deleting the build folder multiple times during my attempts to figure out what was going on and it didn't help. But yes, deleting build is a good thing to try.

**Reply by cromewar** [reply]
If that does not work, another solution is avoid to use `-1` and just deploy the contract on the same execution instead of trying to use the latest one.

**Reply by cromewar** [reply]
@NathanPtrs did you manage to solve the error?

**Reply by NathanPtrs** [reply]
No unfortunately. I can make it work by running the upgrade box script in console immediately after the deploy box script (so they are running on the same execution) and that works for local blockchain. But that is unnecessary with rinkeby where it works fine: I can run deploy and upgrade on separate executions with no issue. But I can't seem to make that work locally.

I appreciate the help though!

**Reply by cromewar** [reply]
Interesting, the problem is given your at the same execution but rinkeby works just fine, this must be something related to brownie itself, I'm glad you have surpassed the error tough.
